hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerManagerImpl.java

```diff
@@ -331,6 +341,10 @@ public void removeContainerReplica(final ContainerID cid,
       throws ContainerNotFoundException, ContainerReplicaNotFoundException {
     if (containerExist(cid)) {
       containerStateManager.removeContainerReplica(cid, replica);
+      // Remove any pending delete replication operations for the deleted
+      // replica.
```

Comment:
one question: what is criteria of the completion of  ecreconstruction  option?
will that be all the reconstructed replicas are seen?

this could be addressed in a separate jira

Reply:
Yea, when the new replicas arrive, they will be reported via ICR / FCR as usual, and then we complete them. It should work exactly the same as for Ratis replicas.